include/hc.hpp

```diff
@@ -2746,14 +2746,25 @@ extern "C" unsigned int __sadhi_u16x2_u8x4(unsigned int src0, unsigned int src1,
 /**
  * Get system timestamp
  */
-extern "C" uint64_t __clock_u64() __HC__;
+extern "C" __attribute__((always_inline))
+uint64_t __ockl_memrealtime_u64(void);
+
+extern "C" inline __attribute((always_inline)) uint64_t __clock_u64() __HC__ {
+  return (uint64_t) __ockl_memrealtime_u64();
```

Comment:
I'm confused as to why you're explicitly casting to `uint64_t` the return value of a function that you declare above as returning `uint64_t`, when returning said value from a function that returns `uint64_t`. Perhaps this is a smidgeon too defensive?